src/wms/src/main/java/org/geoserver/wms/featureinfo/AbstractVectorLayerIdentifier.java

```diff
@@ -42,4 +50,12 @@ public boolean canHandle(MapLayerInfo layer) {
         }
         return result;
     }
+
+    public FeatureSource<? extends FeatureType, ? extends Feature> handleClipParam(
+            FeatureInfoRequestParameters params,
+            FeatureSource<? extends FeatureType, ? extends Feature> featureSource) {
+        Geometry clipGeom = ClipWMSGetMapCallBack.getClipGeometry(params.getGetMapRequest());
```

Comment:
Isn't the clip geometry already parsed and present in the GetMapRequest object? As said yesterday, please move the getClipGeometry method inside the KVP parser, make it private, and use the already parsed geometry available in GetMapRequest everywhere else...

Reply:
understood, guess this call was missed